ecs_fargate/README.md

```diff
@@ -41,11 +40,90 @@ The ECS Fargate check is packaged with the Agent, [run the Agent][1] with your c
 
     This driver collects logs generated by the container and sends them to CloudWatch directly.
 
-3. Finally, [use a lambda function in order to collect logs from CloudWatch and send them to Datadog][10].
+3. Finally, use a [Lambda function][10] to collect logs from CloudWatch and send them to Datadog.
 
 ### Validation
 
-[Run the Agent's `status` subcommand][1] and look for `ecs_fargate` under the Checks section.
+[Run the Datadog Agent's `status` subcommand][1] and look for `ecs_fargate` under the Checks section.
```

Comment:
This part is under the agent check section:
https://github.com/DataDog/integrations-core/blob/master/ecs_fargate/datadog_checks/ecs_fargate/data/conf.yaml.example

I was under the impression there are 2 separate setups.
1. Setup an agent check for ECS Fargate on an agent that is not actually running in ECS Fargate
2. Setup the agent to run in ECS Fargate

If that isn't true, should we remove the first section?

Reply:
The `ecs_fargate` check can only run in an agent container that is running in ECS Fargate. The API it queries cannot be accessed from the outside. It is only reachable from inside a Fargate task.

The two Fargate-specific features of the agent are:

- the ecs_fargate python check
- the autodiscovery module that can discover fargate containers running *in the same task* as the agent. This is why the agent needs to be scheduled as a side car of the application. We can't use the usual "one agent per host" scenario as fargate abstracts the host away from us.

Both these steps are taken care of by [this script](https://github.com/DataDog/datadog-agent/blob/6a0668670fe37784b09073556fd198027e477e2d/Dockerfiles/agent/entrypoint/50-ecs.sh) if the user sets the `ECS_FARGATE` environment variable to true in the container definition while defining their ECS task.

Looking at this readme in its entirety, I realize this part is not useful: https://github.com/DataDog/integrations-core/blob/d09c8a984b31eb031d9a2eaad7504f74110e6994/ecs_fargate/README.md#installation we should point to how to install the agent in ecs fargate instead, and if we don't have such a page we should add the info to the ecs installation page as it's the same thing, except the agent needs to be a side car and not scheduled as a daemon, and we need to add a single env variable to its container definition compared to standard ECS.

This part https://github.com/DataDog/integrations-core/blob/d09c8a984b31eb031d9a2eaad7504f74110e6994/ecs_fargate/README.md#metric-collection is also not accurate, we can't restart the agent in fargate. What we can say about metric collection is that if they installed the agent as we recommend, the ecs_fargate check is running and collecting metrics, and that autodiscovery is enabled, and they can use it by adding labels to their other containers *in the same task*.

The log collection part is great.

My previous comment still applies to the validation part.

This part https://github.com/DataDog/integrations-core/blob/d09c8a984b31eb031d9a2eaad7504f74110e6994/ecs_fargate/README.md#installation-1 is accurate but not complete, we need to explain that this sample task definition needs to be combined with their own application task definition to have the agent run as a side car in it. In fargate, a datadog task is useless because it cannot monitor other tasks.

Finally, the part about service checks is wrong, there is a service check reported by the fargate check, and it's called `fargate_check`: https://github.com/DataDog/integrations-core/blob/d09c8a984b31eb031d9a2eaad7504f74110e6994/ecs_fargate/datadog_checks/ecs_fargate/ecs_fargate.py#L46

Ping me on slack if you need more info.

Reply:
Thanks @hkaj! Updates made based on your comments. Let me know if anything else needs to change.